src/execution.ts

```diff
@@ -395,83 +444,135 @@ function compileOperation(context: CompilationContext) {
-      const nodeBody = compileType(
-        subContext,
-        parentType,
-        fieldType,
-        fieldNodes,
-        [fieldName],
-        [`${GLOBAL_PARENT_NAME}.${name}`],
-        responsePath
-      );
-      const resolverName = getResolverName(parentType.name, fieldName);
-      const topLevelArgs = getArgumentsVarName(resolverName);
-      const validArgs = getValidArgumentsVarName(resolverName);
-      body += `
-  if (${SAFETY_CHECK_PREFIX}${index}) {
-   ${getArguments(
-     subContext,
-     args,
-     topLevelArgs,
-     validArgs,
-     fieldType,
-     responsePath
-   )}
-   if (${validArgs} === true) {
-     ${GLOBAL_EXECUTOR_NAME}(() => ${resolverName}(
-     ${originPaths.join(".")},
-     ${topLevelArgs},
-     ${GLOBAL_CONTEXT_NAME},
-     ${getExecutionInfo(
-       subContext,
-       parentType,
-       fieldType,
-       fieldName,
-       fieldNodes,
-       responsePath
-     )}),
-     (${GLOBAL_PARENT_NAME}, ${fieldName}, err) => {
-      if (err != null) {
-          ${
-            isNonNullType(fieldType)
-              ? GLOBAL_NULL_ERRORS_NAME
-              : GLOBAL_ERRORS_NAME
-          }.push(${
-        createErrorObject(
-          context,
-          fieldNodes,
-          responsePath,
-          "err.message != null ? err.message : err",
-          "err"
-        ) /* TODO: test why no return here*/
-      });
+  context.deferred.forEach((deferredField, index) => {
+    body += `
+      if (${SAFETY_CHECK_PREFIX}${index}) {
+        ${compileDeferredField(context, deferredField)}
+      }`;
+  });
+  return body;
+}
+
+function compileDeferredField(
+  context: CompilationContext,
+  deferredField: DeferredField,
+  appendix?: string
+): string {
+  const {
+    name,
+    originPaths,
+    destinationPaths,
+    fieldNodes,
+    fieldType,
+    fieldName,
+    responsePath,
+    parentType,
+    args
+  } = deferredField;
+  const subContext = createSubCompilationContext(context);
+  const nodeBody = compileType(
+    subContext,
+    parentType,
+    fieldType,
+    fieldNodes,
+    [fieldName],
+    [`${GLOBAL_PARENT_NAME}.${name}`],
+    responsePath
+  );
+  const parentIndexes = getParentIndexes(context);
+  const resolverName = getResolverName(parentType.name, fieldName);
+  const resolverHandler = getHoistedFunctionName(
+    context,
+    `${name}${resolverName}Handler`
+  );
+  const topLevelArgs = getArgumentsVarName(resolverName);
+  const validArgs = getValidArgumentsVarName(resolverName);
+  const executionError = createErrorObject(
+    context,
+    fieldNodes,
+    responsePath,
+    "err.message != null ? err.message : err",
+    "err"
+  );
+  const executionInfo = getExecutionInfo(
+    subContext,
+    parentType,
+    fieldType,
+    fieldName,
+    fieldNodes,
+    responsePath
+  );
+  const emptyError = createErrorObject(context, fieldNodes, responsePath, '""');
+  const resolverParentObject = originPaths.join(".");
```

Comment:
Since the result of `.join()` is a string, I would use `resolverParent` or `resolverParentPath`